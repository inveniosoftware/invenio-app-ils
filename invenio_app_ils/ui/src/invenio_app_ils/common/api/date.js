import { DateTime } from 'luxon';
import _get from 'lodash/get';

export const fromISO = stringDate => {
  return DateTime.fromISO(stringDate);
};

/**
 * Converts datetime JSON string to luxon Datetime object
 * @param date
 * @returns {DateTime}
 */
export const toISO = date => {
  return date.toISO();
};

/**
 *  Serializes date for display in short format
 *  @param {DateTime} date luxon DateTime
 *  @re
 */
export const toShortDateTime = date => {
  return date.toFormat('yyyy-MM-dd HH:mm');
};

/**
 *  Serializes date for display in short format
 *  @param {DateTime} date luxon DateTime
 *  @re
 */
export const toShortDate = date => {
  return date.toFormat('yyyy-MM-dd');
};

/**
 *  Serializes date for UTC date in short format used to
 *  normalise dates for REST API
 *  @param {DateTime} date luxon DateTime
 *  @re
 */
export const toUTCShortDate = date => {
  if (!(typeof date === 'string' || date instanceof DateTime)) {
    throw Error('Wrong date type provided');
  }
  if (typeof date === 'string') {
    date = fromISO(date);
  }
  return date.toUTC().toFormat('yyyy-MM-dd');
};

/**
 *  Serializes date for display in tables
 */
export function dateFormatter({ col, row }) {
  const dateField = _get(row, col.field);
  return dateField ? toShortDate(fromISO(dateField)) : null;
}
