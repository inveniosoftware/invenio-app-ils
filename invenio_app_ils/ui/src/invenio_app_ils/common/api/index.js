export { document } from './documents/document';
export { http } from './base';
export { item } from './items/item';
export { eitem } from './eitems/eitem';
export { loan } from './loans/loan';
export { patron } from './patrons/patron';
export { location } from './locations/location';
export { internalLocation } from './locations/internalLocation';
export { keyword } from './keywords/keyword';
export { series } from './series/series';
export { related } from './related/related';
