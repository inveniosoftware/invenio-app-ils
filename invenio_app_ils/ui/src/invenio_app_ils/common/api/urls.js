export const ApiURLS = {
  loans: {
    loan: '/circulation/loans/:loanPid',
    list: '/circulation/loans/',
    checkout: '/circulation/loans/checkout',
    request: '/circulation/loans/request',
  },
  items: {},
  documents: {},
  locations: {},
  users: {},
};
