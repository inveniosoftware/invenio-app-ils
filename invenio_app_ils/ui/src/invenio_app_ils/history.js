import { createBrowserHistory } from 'history';

const history = createBrowserHistory();

export const goToHandler = path => e => {
  history.push(path);
};

export const goTo = (path, state = {}) => {
  history.push(path, state);
};

export default history;
