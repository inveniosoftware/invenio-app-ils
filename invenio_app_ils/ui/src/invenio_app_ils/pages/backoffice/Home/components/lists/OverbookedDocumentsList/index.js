import { connect } from 'react-redux';
import { fetchOverbookedDocuments } from './state/actions';
import OverbookedDocumentsListComponent from './OverbookedDocumentsList';
import { compose } from 'redux';
import { withRouter } from 'react-router-dom';

const mapStateToProps = state => ({
  data: state.overbookedDocuments.data,
  error: state.overbookedDocuments.error,
  isLoading: state.overbookedDocuments.isLoading,
  hasError: state.overbookedDocuments.hasError,
});

const mapDispatchToProps = dispatch => ({
  fetchOverbookedDocuments: () => dispatch(fetchOverbookedDocuments()),
});

export const OverbookedDocumentsList = compose(
  withRouter,
  connect(
    mapStateToProps,
    mapDispatchToProps
  )
)(OverbookedDocumentsListComponent);
