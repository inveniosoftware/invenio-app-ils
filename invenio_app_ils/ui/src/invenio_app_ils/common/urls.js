import { generatePath } from 'react-router-dom';
import { loan as loanApi } from './api/loan';
import { item as itemApi } from './api/item';

export const FrontSiteURLS = {};

const BackOfficeBase = '/backoffice';

export const BackOfficeURLS = {
  home: BackOfficeBase,
  documentDetails: `${BackOfficeBase}/documents/:documentPid`,
  itemsSearch: `${BackOfficeBase}/items`,
  itemDetails: `${BackOfficeBase}/items/:itemPid`,
  loansSearch: `${BackOfficeBase}/loans`,
  loanDetails: `${BackOfficeBase}/loans/:loanPid`,
  patronDetails: `${BackOfficeBase}/users/:userPid`,
};

export const viewLoanDetailsUrl = loanPid => {
  return generatePath(BackOfficeURLS.loanDetails, {
    loanPid: loanPid,
  });
};

export const showAllLoansUrl = (
  documentPid,
  itemPid,
  state,
  patronPid,
  extraQuery
) => {
  const qs = loanApi.buildLoansQuery(
    documentPid,
    itemPid,
    state,
    patronPid,
    extraQuery
  );
  return `${BackOfficeURLS.loansSearch}?q=${qs}`;
};

export const viewItemDetailsUrl = itemPid => {
  return generatePath(BackOfficeURLS.itemDetails, {
    itemPid: itemPid,
  });
};

export const showAllItemsUrl = (documentPid, state, extraQuery) => {
  const qs = itemApi.buildItemsQuery(documentPid, state, extraQuery);
  return `${BackOfficeURLS.itemsSearch}?q=${qs}`;
};
