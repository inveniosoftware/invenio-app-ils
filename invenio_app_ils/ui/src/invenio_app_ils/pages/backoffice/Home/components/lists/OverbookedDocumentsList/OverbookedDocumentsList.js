import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { generatePath } from 'react-router';
import { Loader, Error } from '../../../../../../common/components';
import { ResultsTable } from '../../../../../../common/components';
import { document as documentApi } from '../../../../../../common/api';
import {
  BackOfficeURLS,
  documentsSearchQueryUrl,
} from '../../../../../../common/urls';
import { formatter } from '../../../../../../common/components/ResultsTable/formatters';
import { SeeAllButton } from '../../../../components/buttons';

export default class OverbookedDocumentsList extends Component {
  constructor(props) {
    super(props);
    this.fetchOverbookedDocuments = props.fetchOverbookedDocuments;
    this.showDetailsUrl = BackOfficeURLS.documentDetails;
    this.seeAllUrl = documentsSearchQueryUrl;
  }

  componentDidMount() {
    this.fetchOverbookedDocuments();
  }

  _showDetailsHandler = document_pid =>
    this.props.history.push(
      generatePath(this.showDetailsUrl, { documentPid: document_pid })
    );

  _seeAllButton = () => {
    const _click = () =>
      this.props.history.push(
        this.seeAllUrl(
          documentApi
            .query()
            .overbooked()
            .qs()
        )
      );

    return <SeeAllButton clickHandler={() => _click()} />;
  };

  prepareData(data) {
    return data.hits.map(row => formatter.document.toTable(row));
  }

  _render_table(data) {
    const rows = this.prepareData(data);
    rows.totalHits = data.total;
    return (
      <ResultsTable
        rows={rows}
        title={'Overbooked documents'}
        subtitle={
          'Documents with more requests than the number of available items for loan.'
        }
        rowActionClickHandler={this._showDetailsHandler}
        seeAllComponent={this._seeAllButton()}
        showMaxRows={this.props.showMaxEntries}
        fixed
        singleLine
      />
    );
  }

  render() {
    const { data, isLoading, error } = this.props;
    return (
      <Loader isLoading={isLoading}>
        <Error error={error}>{this._render_table(data)}</Error>
      </Loader>
    );
  }
}

OverbookedDocumentsList.propTypes = {
  fetchOverbookedDocuments: PropTypes.func.isRequired,
  data: PropTypes.object.isRequired,
  showMaxEntries: PropTypes.number,
};

OverbookedDocumentsList.defaultProps = {
  showMaxEntries: 5,
};
