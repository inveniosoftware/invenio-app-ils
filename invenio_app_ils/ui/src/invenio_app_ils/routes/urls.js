import { generatePath } from 'react-router-dom';
import { invenioConfig } from '../common/config';

const FrontSiteBase = '/';

const FrontSiteRoutesList = {
  home: FrontSiteBase,
  patronProfile: `${FrontSiteBase}profile`,
  documentsList: `${FrontSiteBase}search`,
  documentDetails: `${FrontSiteBase}records/:documentPid`,
};

const FrontSiteRoutesGenerators = {
  documentsListWithQuery: qs => `${FrontSiteRoutesList.documentsList}?q=${qs}`,
  documentDetailsFor: documentPid =>
    generatePath(FrontSiteRoutesList.documentDetails, {
      documentPid: documentPid,
    }),
};

export const FrontSiteRoutes = {
  ...FrontSiteRoutesGenerators,
  ...FrontSiteRoutesList,
};

const BackOfficeBase = '/backoffice';

const BackOfficeRoutesList = {
  home: BackOfficeBase,
  documentsList: `${BackOfficeBase}/documents`,
  documentsCreate: `${BackOfficeBase}/documents/create`,
  documentDetails: `${BackOfficeBase}/documents/:documentPid`,
  eitemsList: `${BackOfficeBase}/eitems`,
  eitemDetails: `${BackOfficeBase}/eitems/:eitemPid`,
  itemsList: `${BackOfficeBase}/items`,
  itemDetails: `${BackOfficeBase}/items/:itemPid`,
  loansList: `${BackOfficeBase}/loans`,
  loanDetails: `${BackOfficeBase}/loans/:loanPid`,
  patronsList: `${BackOfficeBase}/patrons`,
  patronDetails: `${BackOfficeBase}/patrons/:patronPid`,
  locationsList: `${BackOfficeBase}/locations`,
  seriesList: `${BackOfficeBase}/series`,
  seriesDetails: `${BackOfficeBase}/series/:seriesPid`,
};

const BackOfficeRouteGenerators = {
  documentsListWithQuery: qs => `${BackOfficeRoutesList.documentsList}?q=${qs}`,
  documentDetailsFor: documentPid =>
    generatePath(BackOfficeRoutesList.documentDetails, {
      documentPid: documentPid,
    }),
  eitemDetailsFor: eitemPid =>
    generatePath(BackOfficeRoutesList.eitemDetails, { eitemPid: eitemPid }),
  itemsListWithQuery: qs => `${BackOfficeRoutesList.itemsList}?q=${qs}`,
  itemDetailsFor: itemPid =>
    generatePath(BackOfficeRoutesList.itemDetails, { itemPid: itemPid }),
  loansListWithQuery: qs => `${BackOfficeRoutesList.loansList}?q=${qs}`,
  loanDetailsFor: loanPid =>
    generatePath(BackOfficeRoutesList.loanDetails, { loanPid: loanPid }),
  patronDetailsFor: patronPid =>
    generatePath(BackOfficeRoutesList.patronDetails, { patronPid: patronPid }),
  seriesListWithQuery: qs => `${BackOfficeRoutesList.seriesList}?q=${qs}`,
  seriesDetailsFor: seriesPid =>
    generatePath(BackOfficeRoutesList.seriesDetails, { seriesPid: seriesPid }),
};

export const BackOfficeRoutes = {
  ...BackOfficeRoutesList,
  ...BackOfficeRouteGenerators,
};

export const openRecordEditor = (path, recid = '') => {
  window.open(`${invenioConfig.editor.url}${path}${recid}`, '_blank');
};
