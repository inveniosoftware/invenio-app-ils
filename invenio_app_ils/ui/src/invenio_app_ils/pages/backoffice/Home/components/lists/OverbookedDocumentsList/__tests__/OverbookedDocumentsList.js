import React from 'react';
import { shallow, mount } from 'enzyme';
import { Settings } from 'luxon';
import { fromISO } from '../../../../../../../common/api/date';
import { BackOfficeRoutes } from '../../../../../../../routes/urls';
import OverbookedDocumentsList from '../OverbookedDocumentsList';
import history from '../../../../../../../history';

jest.mock('../../../../../../../common/config');

Settings.defaultZoneName = 'utc';
const stringDate = fromISO('2018-01-01T11:05:00+01:00');

describe('OverbookedDocumentsList tests', () => {
  let component;
  afterEach(() => {
    if (component) {
      component.unmount();
    }
  });

  it('should load the details component', () => {
    const component = shallow(
      <OverbookedDocumentsList
        data={{ hits: [], total: 0 }}
        fetchOverbookedDocuments={() => {}}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should fetch documents on mount', () => {
    const mockedFetchLoans = jest.fn();
    component = mount(
      <OverbookedDocumentsList
        data={{ hits: [], total: 0 }}
        fetchOverbookedDocuments={mockedFetchLoans}
      />
    );
    expect(mockedFetchLoans).toHaveBeenCalled();
  });

  it('should render show a message with no documents', () => {
    component = mount(
      <OverbookedDocumentsList
        data={{ hits: [], total: 0 }}
        fetchOverbookedDocuments={() => {}}
      />
    );

    expect(component).toMatchSnapshot();
    const message = component
      .find('Message')
      .filterWhere(element => element.prop('data-test') === 'no-results');
    expect(message).toHaveLength(1);
  });

  it('should render documents', () => {
    const data = {
      hits: [
        {
          id: 1,
          updated: stringDate,
          created: stringDate,
          pid: 'doc1',
          metadata: {
            title: 'X',
            authors: ['Author1'],
            abstracts: 'This is an abstract',
            circulation: {
              pending_loans: 1,
              has_items_for_loan: 2,
            },
          },
        },
        {
          id: 2,
          updated: stringDate,
          created: stringDate,
          pid: 'doc2',
          metadata: {
            title: 'X',
            authors: ['Author1'],
            abstracts: 'This is an abstract',
          },
        },
      ],
      total: 2,
    };

    component = mount(
      <OverbookedDocumentsList
        data={data}
        fetchOverbookedDocuments={() => {}}
      />
    );

    expect(component).toMatchSnapshot();
    const rows = component
      .find('TableRow')
      .filterWhere(
        element =>
          element.prop('data-test') === 'doc1' ||
          element.prop('data-test') === 'doc2'
      );
    expect(rows).toHaveLength(2);

    const footer = component
      .find('TableRow')
      .filterWhere(element => element.prop('data-test') === 'footer');
    expect(footer).toHaveLength(0);
  });

  it('should go to loan details when clicking on a document', () => {
    const mockedHistoryPush = jest.fn();
    history.push = mockedHistoryPush;
    const data = {
      hits: [
        {
          id: 2,
          updated: stringDate,
          created: stringDate,
          pid: 'doc2',
          metadata: {
            title: 'X',
            authors: ['Author1'],
            abstracts: 'This is an abstract',
          },
        },
      ],
      total: 1,
    };

    component = mount(
      <OverbookedDocumentsList
        data={data}
        fetchOverbookedDocuments={() => {}}
        showMaxEntries={1}
      />
    );

    const firstId = data.hits[0].pid;
    const button = component
      .find('TableRow')
      .filterWhere(element => element.prop('data-test') === firstId)
      .find('i');
    button.simulate('click');

    const expectedParam = BackOfficeRoutes.documentDetailsFor(firstId);
    expect(mockedHistoryPush).toHaveBeenCalledWith(expectedParam, {});
  });
});
