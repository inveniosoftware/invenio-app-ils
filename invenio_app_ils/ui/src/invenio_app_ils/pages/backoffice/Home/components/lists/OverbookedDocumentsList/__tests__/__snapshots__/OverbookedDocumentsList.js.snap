// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverbookedDocumentsList tests should load the details component 1`] = `
<Loader>
  <Error>
    <ResultsTable
      fixed={true}
      headerActionClickHandler={null}
      headerActionComponent={null}
      rowActionClickHandler={[Function]}
      rows={Array []}
      seeAllComponent={
        <SeeAllButton
          clickHandler={[Function]}
          disabled={false}
          fluid={false}
        />
      }
      showMaxRows={5}
      singleLine={true}
      subtitle="Documents with more requests than the number of available items for loan."
      title="Overbooked documents"
    />
  </Error>
</Loader>
`;

exports[`OverbookedDocumentsList tests should render documents 1`] = `
<OverbookedDocumentsList
  data={
    Object {
      "hits": Array [
        Object {
          "ID": "doc1",
          "circulation": Object {
            "loanableItems": 2,
            "pendingLoans": 1,
          },
          "created": "2018-01-01T10:05:00.000Z",
          "document_pid": "doc1",
          "title": "X",
          "updated": "2018-01-01T10:05:00.000Z",
        },
        Object {
          "ID": "doc2",
          "created": "2018-01-01T10:05:00.000Z",
          "document_pid": "doc2",
          "title": "X",
          "updated": "2018-01-01T10:05:00.000Z",
        },
      ],
      "total": 2,
    }
  }
  fetchOverbookedDocuments={[Function]}
  history={[Function]}
  showMaxEntries={5}
>
  <Loader>
    <Error>
      <ResultsTable
        fixed={true}
        headerActionClickHandler={null}
        headerActionComponent={null}
        rowActionClickHandler={[Function]}
        rows={
          Array [
            Object {
              "ID": "doc1",
              "Loan requests": undefined,
              "Loanable items": undefined,
              "Title": "X",
            },
            Object {
              "ID": "doc2",
              "Title": "X",
            },
          ]
        }
        seeAllComponent={
          <SeeAllButton
            clickHandler={[Function]}
            disabled={false}
            fluid={false}
          />
        }
        showMaxRows={5}
        singleLine={true}
        subtitle="Documents with more requests than the number of available items for loan."
        title="Overbooked documents"
      >
        <Segment>
          <div
            className="ui segment"
          >
            <Grid>
              <div
                className="ui grid"
              >
                <GridRow>
                  <div
                    className="row"
                  >
                    <GridColumn
                      verticalAlign="middle"
                      width={13}
                    >
                      <div
                        className="middle aligned thirteen wide column"
                      >
                        <Header
                          as="h3"
                          content="Overbooked documents"
                          subheader="Documents with more requests than the number of available items for loan."
                        >
                          <h3
                            className="ui header"
                          >
                            Overbooked documents
                            <HeaderSubheader
                              content="Documents with more requests than the number of available items for loan."
                            >
                              <div
                                className="sub header"
                              >
                                Documents with more requests than the number of available items for loan.
                              </div>
                            </HeaderSubheader>
                          </h3>
                        </Header>
                      </div>
                    </GridColumn>
                    <GridColumn
                      textAlign="right"
                      width={3}
                    >
                      <div
                        className="right aligned three wide column"
                      />
                    </GridColumn>
                  </div>
                </GridRow>
              </div>
            </Grid>
            <Table
              as="table"
              fixed={true}
              selectable={true}
              singleLine={true}
              striped={true}
            >
              <table
                className="ui fixed selectable single line striped table"
              >
                <ResultsTableHeader
                  columns={
                    Array [
                      "ID",
                      "Title",
                      "Loan requests",
                      "Loanable items",
                    ]
                  }
                  withRowAction={true}
                >
                  <TableHeader
                    as="thead"
                  >
                    <thead
                      className=""
                    >
                      <TableRow
                        as="tr"
                        cellAs="td"
                        data-test="header"
                      >
                        <tr
                          className=""
                          data-test="header"
                        >
                          <TableHeaderCell
                            as="th"
                            collapsing={true}
                            width={2}
                          >
                            <TableCell
                              as="th"
                              className=""
                              collapsing={true}
                              width={2}
                            >
                              <th
                                className="collapsing two wide"
                              />
                            </TableCell>
                          </TableHeaderCell>
                          <TableHeaderCell
                            as="th"
                            key="ID"
                            width={2}
                          >
                            <TableCell
                              as="th"
                              className=""
                              width={2}
                            >
                              <th
                                className="two wide"
                              >
                                ID
                              </th>
                            </TableCell>
                          </TableHeaderCell>
                          <TableHeaderCell
                            as="th"
                            key="Title"
                          >
                            <TableCell
                              as="th"
                              className=""
                            >
                              <th
                                className=""
                              >
                                Title
                              </th>
                            </TableCell>
                          </TableHeaderCell>
                          <TableHeaderCell
                            as="th"
                            key="Loan requests"
                          >
                            <TableCell
                              as="th"
                              className=""
                            >
                              <th
                                className=""
                              >
                                Loan requests
                              </th>
                            </TableCell>
                          </TableHeaderCell>
                          <TableHeaderCell
                            as="th"
                            key="Loanable items"
                          >
                            <TableCell
                              as="th"
                              className=""
                            >
                              <th
                                className=""
                              >
                                Loanable items
                              </th>
                            </TableCell>
                          </TableHeaderCell>
                        </tr>
                      </TableRow>
                    </thead>
                  </TableHeader>
                </ResultsTableHeader>
                <ResultsTableBody
                  columns={
                    Array [
                      "ID",
                      "Title",
                      "Loan requests",
                      "Loanable items",
                    ]
                  }
                  rowActionClickHandler={[Function]}
                  rows={
                    Array [
                      Object {
                        "ID": "doc1",
                        "Loan requests": undefined,
                        "Loanable items": undefined,
                        "Title": "X",
                      },
                      Object {
                        "ID": "doc2",
                        "Title": "X",
                      },
                    ]
                  }
                >
                  <TableBody
                    as="tbody"
                  >
                    <tbody
                      className=""
                    >
                      <TableRow
                        as="tr"
                        cellAs="td"
                        data-test="doc1"
                        key="doc1"
                      >
                        <tr
                          className=""
                          data-test="doc1"
                        >
                          <TableCell
                            as="td"
                            textAlign="center"
                          >
                            <td
                              className="center aligned"
                            >
                              <Button
                                as="button"
                                circular={true}
                                compact={true}
                                data-test="btn-view-details-doc1"
                                icon="eye"
                                onClick={[Function]}
                              >
                                <button
                                  className="ui circular compact icon button"
                                  data-test="btn-view-details-doc1"
                                  onClick={[Function]}
                                >
                                  <Icon
                                    as="i"
                                    name="eye"
                                  >
                                    <i
                                      aria-hidden="true"
                                      className="eye icon"
                                    />
                                  </Icon>
                                </button>
                              </Button>
                            </td>
                          </TableCell>
                          <TableCell
                            as="td"
                            data-test="ID-doc1"
                            key="0-doc1"
                          >
                            <td
                              className=""
                              data-test="ID-doc1"
                            >
                              doc1
                            </td>
                          </TableCell>
                          <TableCell
                            as="td"
                            data-test="Title-doc1"
                            key="1-doc1"
                          >
                            <td
                              className=""
                              data-test="Title-doc1"
                            >
                              X
                            </td>
                          </TableCell>
                          <TableCell
                            as="td"
                            data-test="Loan requests-doc1"
                            key="2-doc1"
                          >
                            <td
                              className=""
                              data-test="Loan requests-doc1"
                            >
                              -
                            </td>
                          </TableCell>
                          <TableCell
                            as="td"
                            data-test="Loanable items-doc1"
                            key="3-doc1"
                          >
                            <td
                              className=""
                              data-test="Loanable items-doc1"
                            >
                              -
                            </td>
                          </TableCell>
                        </tr>
                      </TableRow>
                      <TableRow
                        as="tr"
                        cellAs="td"
                        data-test="doc2"
                        key="doc2"
                      >
                        <tr
                          className=""
                          data-test="doc2"
                        >
                          <TableCell
                            as="td"
                            textAlign="center"
                          >
                            <td
                              className="center aligned"
                            >
                              <Button
                                as="button"
                                circular={true}
                                compact={true}
                                data-test="btn-view-details-doc2"
                                icon="eye"
                                onClick={[Function]}
                              >
                                <button
                                  className="ui circular compact icon button"
                                  data-test="btn-view-details-doc2"
                                  onClick={[Function]}
                                >
                                  <Icon
                                    as="i"
                                    name="eye"
                                  >
                                    <i
                                      aria-hidden="true"
                                      className="eye icon"
                                    />
                                  </Icon>
                                </button>
                              </Button>
                            </td>
                          </TableCell>
                          <TableCell
                            as="td"
                            data-test="ID-doc2"
                            key="0-doc2"
                          >
                            <td
                              className=""
                              data-test="ID-doc2"
                            >
                              doc2
                            </td>
                          </TableCell>
                          <TableCell
                            as="td"
                            data-test="Title-doc2"
                            key="1-doc2"
                          >
                            <td
                              className=""
                              data-test="Title-doc2"
                            >
                              X
                            </td>
                          </TableCell>
                          <TableCell
                            as="td"
                            data-test="Loan requests-doc2"
                            key="2-doc2"
                          >
                            <td
                              className=""
                              data-test="Loan requests-doc2"
                            >
                              -
                            </td>
                          </TableCell>
                          <TableCell
                            as="td"
                            data-test="Loanable items-doc2"
                            key="3-doc2"
                          >
                            <td
                              className=""
                              data-test="Loanable items-doc2"
                            >
                              -
                            </td>
                          </TableCell>
                        </tr>
                      </TableRow>
                    </tbody>
                  </TableBody>
                </ResultsTableBody>
                <ResultsTableFooter
                  allRowsNumber={2}
                  columnsNumber={4}
                  seeAllComponent={
                    <SeeAllButton
                      clickHandler={[Function]}
                      disabled={false}
                      fluid={false}
                    />
                  }
                  showMaxRows={5}
                />
              </table>
            </Table>
          </div>
        </Segment>
      </ResultsTable>
    </Error>
  </Loader>
</OverbookedDocumentsList>
`;

exports[`OverbookedDocumentsList tests should render show a message with no documents 1`] = `
<OverbookedDocumentsList
  data={
    Object {
      "hits": Array [],
      "total": 0,
    }
  }
  fetchOverbookedDocuments={[Function]}
  history={[Function]}
  showMaxEntries={5}
>
  <Loader>
    <Error>
      <ResultsTable
        fixed={true}
        headerActionClickHandler={null}
        headerActionComponent={null}
        rowActionClickHandler={[Function]}
        rows={Array []}
        seeAllComponent={
          <SeeAllButton
            clickHandler={[Function]}
            disabled={false}
            fluid={false}
          />
        }
        showMaxRows={5}
        singleLine={true}
        subtitle="Documents with more requests than the number of available items for loan."
        title="Overbooked documents"
      >
        <Segment>
          <div
            className="ui segment"
          >
            <Grid>
              <div
                className="ui grid"
              >
                <GridRow>
                  <div
                    className="row"
                  >
                    <GridColumn
                      verticalAlign="middle"
                      width={13}
                    >
                      <div
                        className="middle aligned thirteen wide column"
                      >
                        <Header
                          as="h3"
                          content="Overbooked documents"
                          subheader="Documents with more requests than the number of available items for loan."
                        >
                          <h3
                            className="ui header"
                          >
                            Overbooked documents
                            <HeaderSubheader
                              content="Documents with more requests than the number of available items for loan."
                            >
                              <div
                                className="sub header"
                              >
                                Documents with more requests than the number of available items for loan.
                              </div>
                            </HeaderSubheader>
                          </h3>
                        </Header>
                      </div>
                    </GridColumn>
                    <GridColumn
                      textAlign="right"
                      width={3}
                    >
                      <div
                        className="right aligned three wide column"
                      />
                    </GridColumn>
                  </div>
                </GridRow>
              </div>
            </Grid>
            <Message
              data-test="no-results"
            >
              <div
                className="ui message"
                data-test="no-results"
              >
                There are no 
              </div>
            </Message>
          </div>
        </Segment>
      </ResultsTable>
    </Error>
  </Loader>
</OverbookedDocumentsList>
`;
